### ISSUE TITLE
WARN [runner] Can't process result by diff processor: can't prepare diff by revgrep: could not read git repo: error executing git diff "REV" "": exit status 128 

### ISSUE THREAD

**Issue by mintzhao** [golangci/golangci-lint#161]
WARN [runner] Can't process result by diff processor: can't prepare diff by revgrep: could not read git repo: error executing git diff "REV" "": exit status 128

**Comment by jirfag** [comment]
how do you run `golangci-lint`? do you have `git` installed?

**Comment by jirfag** [comment]
@mbana please attach `.golangci.yml`, `golangci-lint --version` and how do you run `golangci-lint`

**Comment by liubog2008** [comment]
change

```
issues:
  new-from-rev: "REV"
```

to

```
issues:
  new-from-rev: ""
```

in `.golangci.yml`

**Comment by OrangeFlag** [comment]
Try before golangci-lint
```
git fetch REV
```

**Comment by marcusirgens** [comment]
Hello, future travellers who googled something like "golangci-lint error executing git diff". If your issue is something like this:

> I tried adding the `new-from-rev` config to the linter, it doesn't work. Instead, I get the following error message:
> `level=warning msg="[runner] Can't process result by diff processor: can't prepare diff by revgrep: could not read git repo: error executing git diff \"b00bface\" \"\": exit status 128"`

Try changing your `actions/checkout@v3` step so that it does not do a shallow checkout:

```yml
      - uses: actions/checkout@v3
        with:
          fetch-depth: '0'
```

**Comment by amenasria** [comment]
**TL; DR: This doesn't happen only for shallow clones, the `new-from-rev` feature will fail whenever the `git diff` it uses, [in revgrep.go](https://github.com/bradleyfalzon/revgrep/blob/c04006dc3307c8768bda7a33c7c15d1c6f664e14/revgrep.go#L317), fail (even if it finds the commit). And this will fail silently which is IMO not really practical. We should either display the full `git diff` error by default, or have a `verbose`/`debug` flag enabling people to see what the real error is.**  

**More details:** So I had a pretty different experience with this issue. This happened a bit of time after we enabled the golangci-lint `new-from-rev` [feature in our repository](https://github.com/DataDog/datadog-agent/blob/fcb19ce078e7969d285565beec5d374c5fd623e1/.golangci.yml#L68). We didn't get any problem for 2 months until [a random PR](https://github.com/DataDog/datadog-agent/pull/19506) that wasn't modifying any go files broke the linters with:
```
level=warning msg="[runner] Can't process result by diff processor:
can't prepare diff by revgrep: could not read git repo: 
error executing git diff \"f40667d3841c6339be0d00d53e54a4a63f43f11e\" \"\":
exit status 128"
```

We tried a lot of solutions (including the above) thinking the problem was that we only had a shadow clone of the repository. The suspicious thing was that it only happened on our Windows Gitlab CI runners.

After some investigation, I realized that the `git diff` was not failing because it couldn't find the `new-from-rev` commit but because it couldn't process the diff of some troublesome files:
```
> git diff f40667d3841c6339be0d00d53e54a4a63f43f11e
...
error: cannot spawn astexplain: No such files or directory
fatal: unable to read files diff
```
Long story short our runners where missing the astexplain [diff algorithm](https://git-scm.com/docs/gitattributes#_setting_the_internal_diff_algorithm) binary breaking the `git diff`. The troubleshooting was very long and could have been avoided with a debug flag or a full `git diff` error display.

**Comment by amenasria** [comment]
This is a `revgrep.go` issue though, just posting here in case someone stumbles on something similar !
